services/bettersocial/user/BetterSocialUnblockUserV2.js

```diff
@@ -17,11 +27,26 @@ const BetterSocialUnblockUserV2 = async (token, selfUserId, targetUserId) => {
       message: "You can't unblock yourself"
     };
 
+  const targetAnonymousUserId = await UsersFunction.findAnonymousUserId(User, targetUserId);
+  const anonymousBlockFromSign = await isAnonymousUserBlockedFromSignUser(
+    selfUserId,
+    targetAnonymousUserId?.user_id
+  );
   try {
     await sequelize.transaction(async (t) => {
       await UserBlockUserFunction.userUnblock(UserBlockedUser, selfUserId, targetUserId, {
         transaction: t
       });
+
+      if (anonymousBlockFromSign)
+        await UserBlockUserFunction.userUnblock(
+          UserBlockedUser,
+          selfUserId,
+          targetAnonymousUserId?.user_id,
+          {
+            transaction: t
+          }
+        );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This section of code is missing error handling for the case when `UsersFunction.findAnonymousUserId(User, targetUserId)` fails or returns null. This could lead to unexpected behavior or errors later on in the code. Consider adding error handling or checks for null values.
```diff
+  if (!targetAnonymousUserId) {
+    throw new Error('Could not find anonymous user ID');
+  }

   const anonymousBlockFromSign = await isAnonymousUserBlockedFromSignUser(
     selfUserId,
     targetAnonymousUserId?.user_id
   );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->